integration/networking/port_mapping_linux_test.go

```diff
@@ -859,26 +859,41 @@ func TestAccessPublishedPortFromAnotherNetwork(t *testing.T) {
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			serverID := container.Run(ctx, t, c,
-				container.WithName("server"),
-				container.WithCmd("nc", "-lp", "5000"),
-				container.WithExposedPorts("5000/tcp"),
-				container.WithPortMap(nat.PortMap{"5000/tcp": {{HostPort: "5000"}}}),
-				container.WithNetworkMode(servnet))
-			defer c.ContainerRemove(ctx, serverID, containertypes.RemoveOptions{Force: true})
+			// TODO: Figure out why is it flaky and fix the actual issue.
+			retryFlaky(t, 5, func(t *testing.T) bool {
+				serverID := container.Run(ctx, t, c,
+					container.WithName("server"),
+					container.WithCmd("nc", "-lp", "5000"),
+					container.WithExposedPorts("5000/tcp"),
+					container.WithPortMap(nat.PortMap{"5000/tcp": {{HostPort: "5000"}}}),
+					container.WithNetworkMode(servnet))
+				defer c.ContainerRemove(ctx, serverID, containertypes.RemoveOptions{Force: true})
 
-			clientID := container.Run(ctx, t, c,
-				container.WithName("client"),
-				container.WithCmd("/bin/sh", "-c", fmt.Sprintf("echo foobar | nc -w1 %s 5000", tc.daddr)),
-				container.WithNetworkMode(clientnet))
-			defer c.ContainerRemove(ctx, clientID, containertypes.RemoveOptions{Force: true})
+				clientID := container.Run(ctx, t, c,
+					container.WithName("client"),
+					container.WithCmd("/bin/sh", "-c", fmt.Sprintf("echo foobar | nc -w1 %s 5000", tc.daddr)),
+					container.WithNetworkMode(clientnet))
+				defer c.ContainerRemove(ctx, clientID, containertypes.RemoveOptions{Force: true})
 
-			logs := getContainerStdout(t, ctx, c, serverID)
-			assert.Assert(t, is.Contains(logs, "foobar"), "Payload was not received by the server container")
+				logs := getContainerStdout(t, ctx, c, serverID)
+				return strings.Contains(logs, "foobar")
```

Comment:
Would it make sense to do a similar trick as I did in https://github.com/moby/moby/pull/49600, to still use the `is.Contains` (and have it do the formatting etc for us)? It's a bit unfortunate that the function to execute and/or print these is currently internal in gotest.tools; it has some utilities for that; https://github.com/moby/moby/blob/a9c9d5bb2504f9f6d98ac8e7bce677814858b27d/vendor/gotest.tools/v3/internal/assert/result.go#L13-L62

Reply:
Yep, SGTM